plotly/tests/test_optional/test_figure_factory.py

```diff
@@ -529,3 +530,149 @@ def test_dendrogram_colorscale(self):
+                                             "Make sure all entries of "
+                                             "each column are either "
+                                             "numbers or strings.",
+                                tls.FigureFactory.create_scatterplotmatrix,
+                                df)
+
+        df = pd.DataFrame([[1, 2], ['a', 4]])
+
+        self.assertRaisesRegexp(PlotlyError, "Error in dataframe. "
+                                             "Make sure all entries of "
+                                             "each column are either "
+                                             "numbers or strings.",
+                                tls.FigureFactory.create_scatterplotmatrix,
+                                df)
+
+    def test_same_data_in_index(self):
+
+        # check: either all numbers or strings in index column
+        df = pd.DataFrame([['a', 2], [3, 4]], columns=['apple', 'pear'])
+
+        self.assertRaisesRegexp(PlotlyError, "Error in indexing column. "
+                                             "Make sure all entries of each "
+                                             "column are all numbers or "
+                                             "all strings.",
+                                tls.FigureFactory.create_scatterplotmatrix,
+                                df, index='apple')
+
+        df = pd.DataFrame([[1, 2], ['a', 4]], columns=['apple', 'pear'])
+
+        self.assertRaisesRegexp(PlotlyError, "Error in indexing column. "
+                                             "Make sure all entries of each "
+                                             "column are all numbers or "
+                                             "all strings.",
+                                tls.FigureFactory.create_scatterplotmatrix,
+                                df, index='apple')
+
+    def test_valid_palette(self):
+
+        # check: the palette argument is in a acceptable form
+        df = pd.DataFrame([[1, 2, 3], [4, 5, 6], [7, 8, 9]],
+                          columns=['a', 'b', 'c'])
+
+        self.assertRaisesRegexp(PlotlyError, "You must pick a valid "
+                                             "plotly colorscale name.",
+                                tls.FigureFactory.create_scatterplotmatrix,
+                                df, use_theme=True, index='a',
+                                palette='fake_scale')
+
+        self.assertRaisesRegexp(PlotlyError,
+                                "The items of 'palette' must be "
+                                "tripets of the form a,b,c or "
+                                "'rgbx,y,z' where a,b,c belong "
+                                "to the interval 0,1 and x,y,z "
+                                "belong to 0,255.",
+                                tls.FigureFactory.create_scatterplotmatrix,
+                                df, use_theme=True, palette=1, index='c')
```

Comment:
https://github.com/plotly/plotly.py/pull/417/files#r60284049 (all of these can be made a lot more readable)